app/lib/dom.ts

```diff
@@ -0,0 +1,33 @@
+export function isInSubtreeOf(element: Element, target: Element) {
+  let current: Element | null = target;
+  while (current) {
+    if (current === element) return true;
+    current = current.parentElement;
+  }
+  return false;
+}
+
+export function setClassEnabled(element: HTMLElement, className: string, enabled: boolean) {
+  if (enabled) {
+    element.classList.add(className);
+  } else {
+    element.classList.remove(className);
+  }
+}
+
+export function setOrRemoveStyleProperty(
+  element: HTMLElement,
+  property: string,
+  value: string,
+  { important = false } = {}
+) {
+  if (value) {
+    element.style.setProperty(property, value, important ? "important" : undefined);
+  } else {
+    element.style.removeProperty(property);
+  }
+}
+
+export function createSvgElement(name: string) {
```

Comment:
are we creating so many svgs that this needs to be it's own function in lib?

Reply:
It doesn't need to be in `lib`, this is just a side effect of importing it from my personal toolkit. In order to quickly make progress as a freelancer, I keep a "toolkit" of code, from which I basically drag and drop libs, adapt to fit the target language / ESLint etc., then clean up unused bits.

But I also think it makes sense in this case to keep it in lib, because it's pretty generic and could be used anywhere. You could cite the "you ain't gonna need it" engineering principle here, but I think it's just creating more work for yourself down the line if you constantly co-locate utility functions in the places they are used -- you have to remember where the heck you put that function and then do the work of moving it to a shared location.

There were countless times at Google when I wanted to use some code but wound up copy/pasting it because it was located in someone else's directory that I didn't have visibility into, and I've seen tons of other engineers do the same.